### PR TITLE
Added a background texture to ColorPicker

### PR DESCRIPTION
**Pull request by mrcdk**
So we can see how alpha affects the color.

I've also added the same functionality to the presets

For this to work the `icon_mini_checkerboard.svg` needs to be added which I've already done in this other PR https://github.com/djrm/godot-design/pull/6

### CODE REVIEW

File: scene/gui/color_picker.h [hunk 1]
```diff
@@ -79,6 +79,7 @@ class ColorPicker : public BoxContainer {
 	void _update_text_value();
 	void _text_type_toggled();
 	void _sample_draw();
+    void _preset_draw();
```
Comment (Noshyaar) [inline scene/gui/color_picker.h]: Indentation issue. Please use tab.

File: scene/gui/color_picker.cpp [hunk 2]
```diff
@@ -266,7 +253,11 @@ void ColorPicker::_update_text_value() {
 }
 
 void ColorPicker::_sample_draw() {
-	sample->draw_rect(Rect2(Point2(), Size2(uv_edit->get_size().width, sample->get_size().height * 0.95)), color);
+	Rect2 r = Rect2(Point2(), Size2(uv_edit->get_size().width, sample->get_size().height * 0.95));
+	if(color.a < 1) {
```
Comment (Noshyaar) [inline scene/gui/color_picker.cpp]: Clang-format requires a space between if and condition.
`if (color.a < 1) {`

File: scene/gui/color_picker.cpp [hunk 3]
```diff
@@ -175,31 +175,18 @@ void ColorPicker::_update_color() {
...
 
-	Ref<Image> i = memnew(Image(size.x * presets.size(), size.y, false, Image::FORMAT_RGB8, img));
+	preset->draw_texture_rect(get_icon("MiniCheckerboard", "EditorIcons"), full_preset, true);
 
-	Ref<ImageTexture> t;
-	t.instance();
-	t->create_from_image(i);
-	preset->set_texture(t);
+	Rect2 preset_rect = Rect2(Point2(), size);
+	for(int i = 0; i < presets.size(); i++) {
```
Comment (Noshyaar) [inline scene/gui/color_picker.cpp]: Clang-format requires a space after for.
`for (int i = 0; i < presets.size(); i++) {`

**Comment by Noshyaar** [comment]
Very nice addition! Other than some code style issues, LGTM.

I assume it works in light theme too?

**Comment by mrcdk** [comment]
With the light theme it works...kinda? I'm not sure why the image is darker

**Comment by toger5** [comment]
FINALLY!!!
@mrcdk this is because the svgs colors get modified when importing. try to use a slightly different color.. check editor_themes.cpp line 117 following to see which colors get converted to what.

could u also squash you commits:
```
git reset HEAD~2
git commit -a -m "my commit message for Both commits"
```

**Comment by toger5** [comment]
@mrcdk does that also work for the color picker buttons?

**Comment by mrcdk** [comment]
Okay, done @toger5 
It works with the color picker buttons and I found where to add an exeception to the icon color changes

**Comment by toger5** [comment]
@mrcdk damn of course exceptions is much better!! Forgot they exist...

File: scene/gui/color_picker.cpp [hunk 4]
```diff
@@ -175,31 +175,18 @@ void ColorPicker::_update_color() {
...
-				w[ofs + 1] = uint8_t(CLAMP(presets[(int)x / size.x].g * 255.0, 0, 255));
-				w[ofs + 2] = uint8_t(CLAMP(presets[(int)x / size.x].b * 255.0, 0, 255));
-			}
-		}
-	}
+	Rect2 full_preset = Rect2(Point2(), Size2(size.width * presets.size(), size.height));
+	preset->set_custom_minimum_size(full_preset.size);
 
-	Ref<Image> i = memnew(Image(size.x * presets.size(), size.y, false, Image::FORMAT_RGB8, img));
+	preset->draw_texture_rect(get_icon("MiniCheckerboard", "EditorIcons"), full_preset, true);
```
Comment (akien-mga) [inline scene/gui/color_picker.cpp]: Does that work in games? `ColorPicker` is a Control node and can be used outside the editor too.

Comment (toger5) [reply scene/gui/color_picker.cpp]: I think you are right we have to add it in the default theme...

**Comment by mrcdk** [comment]
Well, I goofed the squash thing, I'm gonna close this PR and open a new one again 😫

**Comment by toger5** [comment]
you can also keep this pr, than make another branch. reset this one a little back... rebase to the new (beautifully squashed branch. and do `git push -f`)

**Pull request closed without merging**